### ISSUE TITLE
Bootstrap re-builds everything after "x.py check"

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#130108]
Some time over the last days, this started happening:
```
$ ./x.py test ui --bless -- null
$ ./x.py check miri
$ ./x.py test ui --bless -- null
```
The third invocation should finish immediately, but instead it rebuilds everything. So what is usually a 1s edit-test cycle now takes a full minute.

Cc @rust-lang/bootstrap

**Comment by onur-ozkan** [comment]
I wonder if https://github.com/rust-lang/rust/pull/128871 has anything to do with this.

**Comment by RalfJung** [comment]
Yeah that sounds likely. Configuring the linker sets some env vars and when those differ between check and test builds, everything gets rebuilt.

**Comment by Kobzol** [comment]
I suppose that we could for now just skip the `check` command from https://github.com/rust-lang/rust/pull/128871. I tried the three commands that you have posted, and the last command did not rebuild anything (or rather, everything was fresh). What code should I modify between running these commands to reproduce the behavior that the third command rebuilds a lot of stuff?

**Comment by Kobzol** [comment]
Nevermind, I reproduced it. And the fix with removing check from configuring the linker works. I'll send a PR.

**Comment by onur-ozkan** [comment]
I think excluding `check` is quite good short-term fix.
